text/0000-scaling-use-cases.md

```diff
@@ -0,0 +1,115 @@
+
+The global wallet has an unlimited number of enumerable addresses. Each index will be associated to a user. For instance:
+
+```
+address_index | user
+--------------|-----------
+            0 | Alice
+            1 | Bob
+            2 | Alice
+            3 | Charlie
+            4 | Charlie
+            5 | Dave
+            6 | Eve
+            7 | Alice
+```
+
+Notice that each index belongs to only one user. But the same user may have one or more indexes.
+
+This approach does not scale easily because the global wallet will have to keep track of more and more addresses over time. The more users, the more addresses. The more addresses, the more memory is required.
+
+Another challenge is to scan the transactions of the global wallet. The most common stop criteria for the scanner is the GAP Limit, which stops the scanner when `N` addresses in a row are empty. An address is empty when it has no transactions.
+
+To illustrate the problem, let's say the GAP Limit is three. If Charlie and Dave have not made any deposits, addresses #3, #4, and #5 will be empty, and the wallet will not find the deposit made by Eve in address #6.
+
+A temporary solution might be to increase the GAP Limit, but the issue will likely happen again in the future.
+
+Another solution might be to artificially generate transactions every three users.
+
+<!-- ##### -->
+<!-- ##### -->
+
+## Global Wallet with Derivation Paths
+
+This case is similar to the Global Wallet. But, instead of associating one index per user, it associates one derivation path per user. For instance:
+
+```
+derivation_path | user
+----------------|----------
+  m/44'/0'/0'/0 | Alice
+  m/44'/0'/0'/1 | Bob
+  m/44'/0'/0'/2 | Charlie
+  m/44'/0'/0'/3 | Dave
+  m/44'/0'/0'/4 | Eve
+```
+
+Notice that each user has one, and only one, derivation path.
+
+Each derivation path leads to an unlimited number of enumerable addresses. Thus, each user has addresses #0, #1, #2, #3, and so on. Notice that Alice's address #0 is different from Bob's address #0 because they had different derivation paths.
+
+This approach has a similar scaling issue of the Global Wallet. The wallet will have to keep track of all these derivation paths and their addresses.
+
+Another challenge is to scan the derivation paths. The most common stop criteria for the scanner is the GAP Limit M/N, which stops the scanner when `M` derivation paths in a row are empty. It considers that a given derivation path is empty when its first `N` addresses are empty.
+
+## One Wallet Per User
+
+This case differs from the previous ones because there will be multiple wallets running.
+
+First of all, there are multiple ways to associate a wallet with a user. To illustrate, some ways are: (i) one seed per user, (ii) one global seed with one passphrase per user, or (iii) one global seed with one derivation path per user.
```

Comment:
Isn't option (iii) the same as the last section (Global Wallet with Derivation Paths)?